### PR TITLE
Create add and remove task functions

### LINKED ISSUE #18: Need Insert, Update, Delete operations
PR #17 completes the DB framework, but did not address Insert, Update, or Delete operations. While addressed as part of issue #10, some parts of the implementation would have crossed paths with portions of the app which are still being designed at the time this message is being written.  

Possible workarounds are:
- Prioritize design and implementation of applicable classes
- Build "skeleton" classes with only the DB portions needed for Insert, Update, and Delete ops

### PR DESCRIPTION
**Pull request by KevinKelly25**
This PR adds two functions `addTask` and `removeTask` to the class `Arc`. This PR implements them with the guidelines described [here](https://github.com/smacademic/project-cgkm/wiki/App-Function-Descriptions-(WIP)).

In order to successfully complete these functions I also had to do the following:
- Add a list of tasks in the arc class
- Expanded the constructors to generate `UUID` within the constructor rather then needing to supply them to the constructor. With this @jgrabowski121 also added the UUID module for the generation of UUIDs
- I added very basic try exception handling until we decide how we want to do exception/error handling within the project.
- I made a test file `test_arc.dart` however as [this issue](
https://github.com/tekartik/sqflite/issues/49) in SQFlite github repo states their is currently issues testing with it. Since these functions require SQFlite heavily I decided to just keep the little test setup I did for future reference but suspend trying to implement unit test within this PR. I do believe we should open an issue with this topic and discuss it further. Does everyone agree?

This PR partially addresses #14 but does not implement the ability to add to app. 

fixes #18

### CODE REVIEW

File: client/src/arcplanner/lib/model/task.dart [hunk 1]
```diff
@@ -15,11 +16,17 @@ class Task {
...
+  Task(this._aid, this._title,
+      {description = null, dueDate = null, location = null}) {
+    this._tid = new Uuid().v4();
+    this._description = description;
+    this._dueDate = duedate;
+    this._location = location;
+  }
 
   // Defines a user map.  Helps with moving info betwen the db
   //   and the app
```
Comment (jgrabowski121) [inline client/src/arcplanner/lib/model/task.dart]: I have not completed my review yet, but I noticed a few extra spaces in a few comments. One of them is on line 28 in the task.dart file, I believe we only need one hanging space.
```suggestion
  //  and the app
```
The other lines I noticed this was on lines 78 and 88 of arc.dart but that may have been intentional.

**Review by jgrabowski121 — COMMENTED** [review]
Thank you @KevinKelly25  for finalizing the branch we were working on. With the exception of the spacing issue I mentioned earlier, I have not found any issues. I am unsure if the try-catch implementation will be final as we may want to do more than print the error but I do not see an issue with its current state for the time being.

**Comment by KevinKelly25** [comment]
Thank you @jgrabowski121 for pointing the spacing out. I have fixed many of them. However, there may be more in other files. For ln 78 and lines surronding it, it was on purpose to point out that it belongs to the comments that did not have the space above it. 

I do wonder which we should use as it seems we are all using a different style for comments. I am suggesting one of the two below. 

```
  *Description: The parameters supply needed information to create a Task within
  * the SQLite database with the exception of the ID. This ID needs to be 
  * generated with a UUID. This data is sent to the SQLite database to be
  * inserted into the task table. If successfully inserted the data should be
  * used to create a task object. That task object should then be added to the
  * list of tasks belonging to the Arc which called the instance method.
  *@param title represents the title of the Task
  *@param description is the information that describes the  task objective. 
  *  This is null by default
  *@param dueDate is the date when the task should be completed by.
  *  This is null by default
  *@param location is where the task takes place. This is null by default
  *Return: Returns the created task
```
or 
```
  * Description: The parameters supply needed information to create a Task within
  *  the SQLite database with the exception of the ID. This ID needs to be 
  *  generated with a UUID. This data is sent to the SQLite database to be
  *  inserted into the task table. If successfully inserted the data should be
  *  used to create a task object. That task object should then be added to the
  *  list of tasks belonging to the Arc which called the instance method.
  * @param title represents the title of the Task
  * @param description is the information that describes the  task objective. 
  *   This is null by default
  * @param dueDate is the date when the task should be completed by.
  *   This is null by default
  * @param location is where the task takes place. This is null by default
  * Return: Returns the created task
```
I prefer the second but it doesn't matter much.

**Comment by MatthewChastain** [comment]
As to the immediately previous message about comment spacing, I personally prefer the second as well @KevinKelly25. Other than a few spelling errors and possible variable name conflicts that I will list below, the code looks good guys!

arc.dart:
- The comment on line 12 has a spelling error: "underscrore" instead of "underscore"
- There are multiple instances where "parentarc" is used where I believe we should be using "parentArc" for consistency purposes as "parentArc is used as well in other places

task.dart:
- There are multiple instances of "duedate" where I believe we want "dueDate" for consistency as "dueDate" is used in other places in the code also

user.dart:
- Line 10's comment uses "underscrore" instead of "underscore"
- Line 21 has "betwen" where I think you wanted "between"

**Comment by KevinKelly25** [comment]
Thank you for pointing that out @MatthewChastain, they should be fixed.

**Comment by jgrabowski121** [comment]
I also believe that option two for a comment style would be the best choice

**Comment by KevinKelly25** [comment]
I updated the branch from my modified files to be in line with option 2 from above.

**Review by jrm86 — APPROVED** [review]
I like these changes. Seems like UUID implementation was a lot easier than I first thought. I like the style of the comments with the extra space.

The SQFlite testing [issue](https://github.com/tekartik/sqflite/issues/49) is troubling, but I suppose that's the risk with using a third-party package. Definitely something to keep an eye on, but I agree that we can put it on the back-burner for now.

Approving

**Review by jgrabowski121 — APPROVED** [review]
I believe the PR is ready to merge now, I am approving as well

**Review by MatthewChastain — APPROVED** [review]
Everything looks good! Thanks @KevinKelly25 for addressing those issues.

**Comment by KevinKelly25** [comment]
Thank you for the reviews 👍

**Pull request merged**